.github/workflows/pr-build.yaml

```diff
@@ -149,9 +148,45 @@ jobs:
           PR_NUMBER: ${{ github.event.pull_request.number }}
         run: |
           . ./hack/ci/setup-env.sh
-      - run: make build-app-image-${{ matrix.APP }}
+      - run: |
+          make build-app-image-${{ matrix.APP }}
+          make save-app-image-${{ matrix.APP }}
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.APP }}-${{github.sha}}
+          path: /tmp/${{ matrix.APP }}.tar
+          retention-days: 1
+
+  push-apps:
+    name: Push ${{ matrix.APP }}
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+    needs: [prepare-matrix, build-apps]
+    strategy:
+      matrix: ${{fromJson(needs.prepare-matrix.outputs.matrix-app)}}
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - name: Download artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ matrix.APP }}-${{github.sha}}
+          path: /tmp
+      - name: Load image
+        run: docker load --input /tmp/${{ matrix.APP }}.tar
       - name: Log into registry
         run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+      - name: Setup environment
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+        run: |
+          . ./hack/ci/setup-env.sh
       - run: make push-app-image-${{ matrix.APP }}
```

Comment:
consider adding delete step after pushing
```yaml
      - name: Delete Capact CLI artifact
        uses: geekyeggo/delete-artifact@v1
        with:
          name: ${{ matrix.APP }}-${{github.sha}}
```
I do not see reason why we want to keep it. Currently, we quite often push and build images, so it will be good to GC it earlier than  1day.